doc/book/app_server/luajit_getmetrics.rst

```diff
@@ -0,0 +1,450 @@
+.. _Snap tutorial: https://ujit.readthedocs.io/en/latest/public/tut-snap.html
+
+Note: Although value names are similar to value names in
+`ujit.getmetrics() <https://ujit.readthedocs.io/en/latest/public/ujit-024.html#ujit-getmetrics>`_
+the values are not the same, primarily because many ujit numbers are not monotonic.
+
+Note: Although value names are similar to value names in :ref:`LuaJIT metrics <metrics-luajit>`,
+and the values are exactly the same, misc.getmetrics() is slightly easier
+because there is no need to ‘require’ the misc module.
+
+
+.. _luajit_getmetrics_c:
+
+getmetrics C API
+~~~~~~~~~~~~~~~~
+
+The Lua ``getmetrics()`` function is a wrapper for the C function ``luaM_metrics()``.
+
+C programs may include a header named libmisclib.h.
+The definitions in libmisclib.h include the lines
+``struct luam_Metrics { ... the same names as are described earlier for Lua ... }`` and
+``LUAMISC_API void luaM_metrics(lua_State *L, struct luam_Metrics *metrics);``
+The names of ``struct luam_Metrics`` members are the same as Lua's :ref:`getmetrics table values <luajit_getmetrics_tablevalues>` names.
+The data types of ``struct luam_Metrics`` members are all ``size_t``.
+The ``luaM_metrics()`` function will fill the ``*metrics`` structure with the metrics related to the Lua state
+anchored to the ``L`` coroutine.
+
+**Example with a C program**
+
+Go through the :ref:`C stored procedures <f_c_tutorial-c_stored_procedures>` tutorial.
+Replace the easy.c example with
+
+.. code-block:: c
+
+    #include "module.h"
+    #include <lmisclib.h>
+    
+    int easy(box_function_ctx_t *ctx, const char *args, const char *args_end)
+    {
+      lua_State *ls = luaT_state();
+      struct luam_Metrics m;
+      luaM_metrics(ls, &m);
+      printf("allocated memory = %lu\n", m.gc_allocated);
+      return 0;
+    }
+
+Now when you go back to the client and execute the requests up to and including the line
+``capi_connection:call('easy')``
+you will see that the display is something like
+"allocated memory = 4431950"
+although the number will vary.
+
+.. _luajit_getmetrics_example_1:
+
+Example with gc_strnum, strhash_miss, and strhash_hit
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To track new string object allocations:
+
+.. code-block:: lua
+
+    function f()
+      collectgarbage("collect")
+      local oldm = misc.getmetrics()
+      local table_of_strings = {}
+      for i = 3000, 4000 do table.insert(table_of_strings, tostring(i)) end
+      for i = 3900, 4100 do table.insert(table_of_strings, tostring(i)) end
+      local newm = misc.getmetrics()
+      print("gc_strnum diff = " .. newm.gc_strnum - oldm.gc_strnum)
+      print("strhash_miss diff = " .. newm.strhash_miss - oldm.strhash_miss)
+      print("strhash_hit diff = " .. newm.strhash_hit - oldm.strhash_hit)
+    end
+    f()
+
+The result will probably be:
+"gc_strnum diff = 1100" because we added 1202 strings but 100 were duplicates,
```

Comment:
Shouldn't it be 1101 and 101? 
Do I understand correctly that the duplicates are 3900..4000?